packages/flutter_tools/lib/src/test/test_time_recorder.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../globals.dart' as globals;
+
+class TestTimeRecorder {
+  final List<TestTimeRecord> _phaseRecords = List<TestTimeRecord>.generate(TestTimePhases.values.length, (_) => TestTimeRecord());
+
+  Stopwatch start(TestTimePhases phase) {
+    return _phaseRecords[phase.index].start();
+  }
+
+  void stop(TestTimePhases phase, Stopwatch stopwatch) {
+    _phaseRecords[phase.index].stop(stopwatch);
+  }
+
+  void print() {
+    for(final TestTimePhases phase in TestTimePhases.values) {
+      globals.printTrace(_getPrintStringForPhase(phase));
+    }
+  }
+
+  List<String> getPrintAsListForTesting() {
+    final List<String> result = <String>[];
+    for(final TestTimePhases phase in TestTimePhases.values) {
+      result.add(_getPrintStringForPhase(phase));
+    }
+    return result;
+  }
+
+  String _getPrintStringForPhase(final TestTimePhases phase) {
+    assert(_phaseRecords[phase.index].isDone());
+    return 'Runtime for phase ${phase.name}: ${_phaseRecords[phase.index]}';
+  }
+}
+
+class TestTimeRecord {
+  Duration _combinedRuntime = Duration.zero;
+  final Stopwatch _wallClockRuntime = Stopwatch();
+  int _currentlyRunningCount = 0;
+
+  Stopwatch start() {
```

Comment:
We have a StopwatchFactory pattern that is used by some of the logger classes to make these testable. I think it would be ideal to have unit tests that only exercise the `TestTimeRecorder` and assert that it records time as expected with a fake stopwatch

Reply:
Done.